src/ai/backend/manager/registry.py

```diff
@@ -664,6 +665,26 @@ def __init__(self, config_server, dbpool,
             )
             result = await conn.execute(query)
             resource_policy = await result.first()
+            # per-user package storage information should be stored in following format:
+            # [vFolder ID, User UUID, vFolder Host]
+            # User UUID should be None if vFolder type is group
+            query = (sa.select([keypairs.c.user])
+                       .select_from(keypairs)
+                       .where(keypairs.c.access_key == sess_ctx.access_key))
+            user_uuid = await conn.scalar(query)
+            matched_vfolders = await query_accessible_vfolders(
+                conn, user_uuid,
+                domain_name=sess_ctx.domain_name,
+                allowed_vfolder_types=['user', 'group'],
+                extra_vf_conds=(vfolders.c.name == '.local'))
+            for folder in matched_vfolders:
+                if (folder['group'] is not None and folder['group'] == str(sess_ctx.group_id))\
+                        or package_directory is None:
+                    package_directory = [folder['id'].hex,
+                                         None if folder['group'] is None else user_uuid.hex,
+                                         folder['host']]
```

Comment:
Could you explain how `package_directry` variable is added instead of reusing existing `mount_map`?

Reply:
This was essential when agent manipulated per-user package directory itself. But since manager now controls folder creation, it'll be better to reuse mount_map instead of adding additional variable. I'll fix it at next commit.